### ISSUE TITLE
(AA) Text colour contrast requirements not met

### ISSUE THREAD

**Issue by otis-bath** [open-sdg/open-sdg#1759]
**Medium level (AA)**

The colour contrast ratio does not meet guidelines requiring that the visual presentation of standard text has a colour contrast ratio of at least 4.5:1 and that large text has a colour contrast ratio of at least 3:1.

The map on 3-3-3 is cited as an example however this issue applies to all maps: https://sdgdata.gov.uk/3-3-3/
![image](https://user-images.githubusercontent.com/90769108/182160675-690ac1e3-4110-49cd-b4b1-4eeaae596f15.png)
The grey text fails colour contrast guidelines against the pale blue (#F7FCFD) background. Users with low vision may encounter difficulty reading this text.
![image](https://user-images.githubusercontent.com/90769108/182160801-3973f7bf-be76-489f-9477-1c30721ba51e.png)

For this to pass we would need the labels to be darker i.e. Black 

Further, this grey text labels does not meet the colour contrast requirements on the green overlay, making the labels darker would resolve both. 
![image](https://user-images.githubusercontent.com/90769108/182161885-9e9557fd-7504-411e-b412-28342b575a58.png)

**Comment by brockfanning** [comment]
@otis-bath My interpretation is that this refers to the city names on the map tile imagery - for example Wembley, Wimbledon, Sutton, etc. If so, this is not part of Open SDG - those tile images are coming from an external source (in the UK's case, from Mapbox.com). I'm not sure if Mapbox has any other options that have a higher contrast, but it might be worth checking. The tile imagery can also be removed entirely, if needed.

But before diving into Mapbox options - all that said, is there any chance for saying that these city names are decorative? They do convey actual information of course, but it is not relevant to the data. Ie, the fact that Wimbledon is south of London is not relevant to London having 779 cases (as in the second screenshot above). All of those city names could disappear and the map would be completely usable.

**Comment by otis-bath** [comment]
@brockfanning thanks for your thoughts on this. Its a tough one to be honest, we can explore what mapbox has because what we have right now isn't ideal. The only thing I would say in the city names side of things for the UK is that (to me anyway) the regions of the UK really doesn't mean anything and a lot can be understood by the placement of cities. For example (I'm sure most people Brits know roughly where London is) however if you weren't from the UK and didn't know London was in the South East you might be surprised why that region is so different to others. Same with NW (Manchester and Liverpool) in close proximity. Just a thought and may be worth considering as the cities have a lot of influence on the UK data so would be useful to at least have some labels

**Comment by brockfanning** [comment]
@otis-bath Makes sense to discuss further on a call. But it's safe to treat this is a UK-specific issue instead of an Open SDG issue. Out of the box the Open SDG maps show no tile imagery. In practice, this issue is relevant for any country that decides to add tile imagery, but each country will need to decide for themselves whether their tile imagery meets their accessibility standards.

**Comment by otis-bath** [comment]
@brockfanning to test adjusting the opacity on the maps which we can feedback to DAC

**Comment by otis-bath** [comment]
Sarah and Phill to investigate options - potentially can be removed from 2.1.0 as UK specific

**Comment by otis-bath** [comment]
Brock to investigate use of custom colour set

**Comment by phillipgardiner** [comment]
@brockfanning @otis-bath @LucyGwilliamAdmin I've been looking into a custom color set a bit - which is definitely possible. One confusing element for me is, yes the text against the shades of green are not accessible. However, working on this basis. Do the shades or green have to be accessible against the background colour (e.g. currently grey) too? and also do the different shades of sequential green have to to accessible from one another too (e.g. should each shade be distinguishable from the next)? Thoughts - i might be opening a can of worms here!

**Comment by phillipgardiner** [comment]
@brockfanning @sarah-beasley @LucyGwilliamAdmin @otis-bath 

 I've looked into the color sequencing a bit today. Assuming we want to stick with the green we could use a sequence of:

#c4e1c6
#b0d1b3
#9bc2a1
#87b28f
#74a37c
#60946b
#4d8559
#3a7747
#276836

Where I find this a little bit confusing is that the different shades of green do not pass contrast guidelines against one another (e.g. two consecutive shades). However doing a bit of reading as long as there is a black pixel between each this should not being a problem. Looking at the current maps there does seem to be a black border around the area selected so i'm guessing its fine.

The above sequence of colours does depend on a number of changes to the map in Jawgmap too, these are change:

1. Text to (Black), #000000)
2. Turn off the text halo
3. Water to (Grey) #949494
4. Land to (white), #FFFFFF

These tweaks ensure there is sufficient contrast between the place names and the greens used, but also ensure there is sufficient contrast between land and water.

**Comment by sarah-beasley** [comment]
@brockfanning DAC have confirmed we have done all we can to improve the maps. This is the [link](http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-site-map-background-accessible2/3-3-3/) to the feature branch where the proposed map background changes have been made - darkening the text, changing the water colour and changing the colour scale. Is this down to us on our end to implement now?
